app/controllers/products_controller.rb

```diff
@@ -4,6 +4,10 @@ class ProductsController < ApplicationController
   # GET /products or /products.json
   def index
     @products = Product.all
+    respond_to do |format|
+      format.html 
+      format.json{render :json => @products.to_json(only: [:title], include: {category: {only: [:name]}}) }
```

Comment:
use `json: @products...` instead of `:json => @products...`. this is an old syntax.
fix indentation
simplify this statement.